addons/account/models/account_payment.py

```diff
@@ -880,6 +879,9 @@ def write(self, vals):
         return super().write(vals)
 
     def unlink(self):
+        self.move_id.filtered(lambda m: m.state != 'draft').button_draft()
+        self.move_id.unlink()
```

Comment:
Check `_unlink_or_reverse` instead which also takes care of lock dates and hashes